build/multilocale.js

```diff
@@ -205,16 +205,97 @@ function L10nManager(gaiaDir,
 
       var doc = utils.getDocument(content);
 
-      buildL10nMeta(htmlFile, doc);
+      return buildL10nMeta(htmlFile, doc).then(function() {
+        if (self.localeBasedir) {
+          // Copy resource files into build_stage directory
+          getL10nResources(htmlFile, webapp, doc);
+        }
+      });
+    });
+
+    return Promise.all(localizeHtmlFiles);
```

Comment:
Note that Promise.all is often misleading... If any of the promises in the fail refects, it will immediately resolve, no matter if the next ones are still running and still not resolved!
Here, if we resolve too quickly and don't have processEvent call, we may easily have broken localization!